[EDU-6598][EDU-6630] feat: added react landing and framework page (#1722)
aziontech/docs: src/content/docs/en/pages/devtools/azion-edge-runtime/compatibility/frameworks/nextjs.mdx

  Static Site Generation, SSG, Content-focused
namespace: documentation_products_runtime_frameworks_nextjs
menu_namespace: runtimeMenu
---

import LinkButton from 'azion-webkit/linkbutton';

## About Next.js

Next.js is a powerful framework for building fast, scalable, and modern web applications. It supports static site generation, server-side rendering, API routes, and more, making it a popular choice for both static and dynamic sites. Next.js offers a flexible file-based routing system, built-in image optimization, and a rich plugin ecosystem.

**Key features**:

- Hybrid static & server rendering
- File-system based routing
- API routes for backend logic
- Built-in image optimization
- Incremental static regeneration
- TypeScript support
- Rich plugin and middleware ecosystem
- Strong community and enterprise support

Next.js also allows for Server Side Rendering (SSR) on the Azion Web Platform. For more information about Next.js compatibility [visit our documentation](/en/documentation/devtools/azion-edge-runtime/compatibility/nextjs/).

## Available Templates

| Template name | Use case | Details |
|:-------------|:---------|:-----|

> | Next.js Static Boilerplate 14 | Static/SPA | <LinkButton link="/en/documentation/products/templates/nextjs-static-boilerplate/" label="Learn more" severity="secondary" /> |

Reviewer: ```suggestion
| Next.js Static Boilerplate | Static/SPA | <LinkButton link="/en/documentation/products/templates/nextjs-static-boilerplate/" label="Learn more" severity="secondary" /> |
```

| Next.js Store Ecommerce Template | E-commerce | <LinkButton link="/en/documentation/products/templates/nextjs-store/" label="Learn more" severity="secondary" /> |
| Next.js App Middleware | Middleware Example | <LinkButton link="/en/documentation/products/templates/nextjs-app-midddleware/" label="Learn more" severity="secondary" /> |
| Next.js App Configurations | Config Example | <LinkButton link="/en/documentation/products/templates/nextjs-app-configurations/" label="Learn more" severity="secondary" /> |
| Next.js Pages Configurations | Config Example | <LinkButton link="/en/documentation/products/templates/nextjs-pages-configurations/" label="Learn more" severity="secondary" /> |
| Next.js Pages Middleware | Middleware Example | <LinkButton link="/en/documentation/products/templates/nextjs-pages-middleware/" label="Learn more" severity="secondary" /> |
| Kupingplug Store| E-commerce | <LinkButton link="/en/documentation/products/templates/kupingplug/" label="Learn more" severity="secondary" /> |
| Next.js Stationary Store | E-commerce | <LinkButton link="/en/documentation/products/templates/stationary-ecommerce/" label="Learn more" severity="secondary" /> |

> | Cosmic Next Blog | Blog | <LinkButton link="/en/documentation/products/templates/cosmic-nextjs-blog/" label="Learn more" severity="secondary" /> |

Reviewer: Next.js

description: >-
    Explore how Azion Bundler integrates with Next.js, a flexible React framework for building high-performance full-stack web applications.
permalink: /documentation/products/devtools/azion-edge-runtime/frameworks-compatibility/nextjs/
meta_tags: >-
  Azion, Edge Computing, Azion Bundler framework, Next.js, JavaScript frameworks,
  Static Site Generation, SSG, Content-focused
namespace: documentation_products_runtime_frameworks_nextjs
menu_namespace: runtimeMenu
---

import LinkButton from 'azion-webkit/linkbutton';

## About Next.js

Next.js is a powerful framework for building fast, scalable, and modern web applications. It supports static site generation, server-side rendering, API routes, and more, making it a popular choice for both static and dynamic sites. Next.js offers a flexible file-based routing system, built-in image optimization, and a rich plugin ecosystem.

**Key features**:

- Hybrid static & server rendering
- File-system based routing
- API routes for backend logic
- Built-in image optimization
- Incremental static regeneration
- TypeScript support
- Rich plugin and middleware ecosystem
- Strong community and enterprise support

> Next.js also allows for Server Side Rendering (SSR) on the Azion Web Platform, for more information about Next.js compatibility visit [click here](/en/documentation/devtools/azion-edge-runtime/compatibility/nextjs/)

Reviewer: ```suggestion
Next.js also allows for Server Side Rendering (SSR) on the Azion Web Platform. For more information about Next.js compatibility [visit our documentation](/en/documentation/devtools/azion-edge-runtime/compatibility/nextjs/).
```

| Next.js Store Ecommerce Template | E-commerce | <LinkButton link="/en/documentation/products/templates/nextjs-store/" label="Learn more" severity="secondary" /> |
| Next.js App Middleware | Middleware Example | <LinkButton link="/en/documentation/products/templates/nextjs-app-midddleware/" label="Learn more" severity="secondary" /> |
| Next.js App Configurations | Config Example | <LinkButton link="/en/documentation/products/templates/nextjs-app-configurations/" label="Learn more" severity="secondary" /> |
| Next.js Pages Configurations | Config Example | <LinkButton link="/en/documentation/products/templates/nextjs-pages-configurations/" label="Learn more" severity="secondary" /> |
| Next.js Pages Middleware | Middleware Example | <LinkButton link="/en/documentation/products/templates/nextjs-pages-middleware/" label="Learn more" severity="secondary" /> |
| Kupingplug  Store| E-commerce | <LinkButton link="/en/documentation/products/templates/kupingplug/" label="Learn more" severity="secondary" /> |
| Next.js Stationary Store | E-commerce | <LinkButton link="/en/documentation/products/templates/stationary-ecommerce/" label="Learn more" severity="secondary" /> |

> | Cosmic Next Blog | Blog | <LinkButton link="/en/documentation/products/templates/cosmic-nextjs-blog/" label="Learn more" severity="secondary" /> |

Reviewer: Cosmic Next.js Blog?

| Cosmic Agency | Agency/Starter | <LinkButton link="/en/documentation/products/templates/cosmic-agency/" label="Learn more" severity="secondary" /> |
| Space Jelly Shop | E-commerce | <LinkButton link="/en/documentation/products/templates/space-jelly-shop/" label="Learn more" severity="secondary" /> |

### Deploying a template using Azion Console

1. Login to [Console](https://console.azion.com).
2. Click the **\+ Create** button.
3. Click the **Templates** section.
4. Choose the desired Next.js template.
5. Connect your GitHub Account.
6. Choose a name for your application.
7. Click **Deploy**.

### Deploying an existing project using import from GitHub

2. In [Console](https://console.azion.com) Click the **\+ Create** button.

> 3. Select Import from Github.

Reviewer: ```suggestion
3. Select **Import from Github**.
```

4. Select your repository.
5. Choose an application name.

> 6. Select the Next.js Preset.

Reviewer: ```suggestion
6. Select the **Next.js Preset**.
```

### Answering `no` to local dev

After indicating you don't want to have a local server running, deploy the Next.js project to the edge.

1. Enter `y` to the following interaction, indicating you want to deploy the project:

```sh
   Do you want to deploy your project? (y/N) 
```

2. Install the project dependencies. Input `y` when the interaction prompts:

```sh 
  Do you want to install project dependencies? This may be required to deploy your project (y/N)
```

3. Wait while the project is built and deployed to the Azion Web Platform.

:::note 
Once the deployment is triggered, Azion will open the browser and take the user to a page on Azion Console where the deployment logs and process can be monitored. If it doesn't open automatically, just click on the provided link.
::: 
 
After the deployment is complete, you'll receive a domain to access your Next.js project on the Azion Platform.

Wait a few minutes so the propagation takes place, and then access your application using the provided domain, which should be similar to `https://xxxxxxx.map.azionedge.net`.

> 

Reviewer: ```suggestion
```
